evm-contracts/src/v0.6/examples/VRFD20.sol

```diff
@@ -0,0 +1,149 @@
+        s_houses[16] = "Valeryon";
+        s_houses[17] = "Manderly";
+        s_houses[18] = "Clegane";
+        s_houses[19] = "Glover";
+        s_houses[20] = "Karstark";
+    }
+
+    /**
+     * @notice Requests randomness from a user-provided seed
+     * @dev This is only an example implementation and not necessarily suitable for mainnet.
+     * @dev You must review your implementation details with extreme care.
+     *
+     * @param userProvidedSeed uint256 unpredictable seed
+     * @param roller address of the roller
+     */
+    function rollDice(uint256 userProvidedSeed, address roller) public onlyOwner returns (bytes32 requestId) {
+        require(LINK.balanceOf(address(this)) >= s_fee, "Not enough LINK to pay fee");
+        require(s_rollers[roller] == bytes32(0), "Already rolled");
+        requestId = requestRandomness(s_keyHash, s_fee, userProvidedSeed);
+        s_rollers[roller] = requestId;
+        emit DiceRolled(requestId, roller);
+    }
+
+    /**
+     * @notice Get the house assigned to the player once the address has rolled
+     * @param player address
+     * @return house as a string
+     */
+    function house(address player) public view returns (string memory) {
+        require(s_rollers[player] != bytes32(0), "Dice not rolled");
+        require(s_results[s_rollers[player]] != 0, "Roll in progress");
+        return s_houses[s_results[s_rollers[player]]];
+    }
+
+    /**
+     * @notice Withdraw LINK from this contract.
+     * @param to the address to withdraw LINK to
+     * @param value the amount of LINK to withdraw
+     */
+    function withdrawLINK(address to, uint256 value) public onlyOwner {
+        require(LINK.transfer(to, value), "Not enough LINK");
+    }
+
+    /**
+     * @notice Set the key hash for the oracle
+     *
+     * @param keyHash bytes32
+     */
+    function setKeyHash(bytes32 keyHash) public onlyOwner {
+        s_keyHash = keyHash;
+    }
+
+    /**
+     * @notice Get the current key hash
+     *
+     * @return bytes32
+     */
+    function keyHash() public view returns (bytes32) {
+        return s_keyHash;
+    }
+
+    /**
+     * @notice Set the oracle fee for requesting randomness
+     *
+     * @param fee uint256
+     */
+    function setFee(uint256 fee) public onlyOwner {
+        s_fee = fee;
+    }
+
+    /**
+     * @notice Get the current fee
+     *
+     * @return uint256
+     */
+    function fee() public view returns (uint256) {
+        return s_fee;
+    }
+
+    /**
+     * @notice Callback function used by VRF Coordinator to return the random number
+     * to this contract.
+     * @dev This is where you do something with randomness!
+     * @dev The VRF Coordinator will only send this function verified responses.
+     *
+     * @param requestId bytes32
+     * @param randomness The random result returned by the oracle
+     */
+    function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
```

Comment:
I know we normally push internal methods below public, but this is kind of where the meat of the contract lies. I'd put the constructor and `rollDice` before it, and everything else after. That is also kind of inline with the actual usage work-flow.